src/app/accounts/update-account-custom-fields.ts

```diff
@@ -0,0 +1,48 @@
+import { getAccountsConfig } from "@config"
+
+import { CouldNotFindError } from "@domain/errors"
+import { NoAccountCustomFieldsError } from "@domain/accounts"
+
+import { AccountCustomFieldsRepository, AccountsRepository } from "@services/mongoose"
+
+const { customFields: customFieldsSchema } = getAccountsConfig()
+
+const defaultValues = customFieldsSchema.reduce((acc, val) => {
+  if (val.defaultValue !== undefined) {
+    acc[val.name] = val.defaultValue
+  }
+  return acc
+}, {} as { [k: string]: AccountCustomFieldValues })
+
+export const updateAccountCustomFields = async ({
+  accountId,
+  modifiedByUserId,
+  customFields,
+}: {
+  accountId: AccountId
+  modifiedByUserId: UserId
+  customFields: { [k: string]: AccountCustomFieldValues }
+}): Promise<AccountCustomFields | ApplicationError> => {
+  if (!customFieldsSchema || customFieldsSchema.length <= 0)
+    return new NoAccountCustomFieldsError()
+
+  const accountCustomFieldsRepo = AccountCustomFieldsRepository()
+
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof Error) return account
+
+  const accountCustomFields = await accountCustomFieldsRepo.findById(account.id)
+  const isEmpty = accountCustomFields instanceof CouldNotFindError
+  if (accountCustomFields instanceof Error && !isEmpty) return accountCustomFields
+
+  let data = customFields
+  if (!isEmpty) {
+    data = { ...accountCustomFields.customFields, ...customFields }
```

Comment:
one tricky part here is how to merge the field that are array. I guess here also the current behavior is that we would replace the array with the new value? (and not merge the array recursively?)

Reply:
note: this is the type of behavior that would be good to put in documentation.

Reply:
Do we want to support arrays? I don't think the use case we are aiming for here needs it and it adds a lot of complexity. Like you mention above we don't want to define our own schema definition language and merging semantics.

Reply:
this was an additional request after first implementation. In this case full array is replaced

Reply:
note: there is no need for Array with the current use case. they're ok with one response only instead of an array of response possible.

Reply:
array support was dropped